src/humancellatlas/data/metadata/api.py

```diff
@@ -107,22 +107,72 @@ def __init__(self, entity: LinkedEntity, other_entity: Entity, forward: bool) ->
                          other_entity.address)
 
 
+@dataclass(init=False, unsafe_hash=True)
```

Comment:
I would have preferred freezing instances and using the implicit and safe __hash__ implementation. Did that not work? If not, why?